src/commands/deploy/getDeployNode.ts

```diff
@@ -26,7 +26,7 @@ export async function getDeployNode(context: IDeployContext, target: Uri | strin
     let isNewWebApp: boolean = isTargetNewWebApp;
     if (target instanceof SiteTreeItemBase) {
         node = target;
-    } else if (defaultWebAppId && defaultWebAppId !== none) {
+    } else if (defaultWebAppId && defaultWebAppId !== none && !(ext.azureAccountTreeItem.trialAppNode && ext.azureAccountTreeItem.isLoggedIn)) {
```

Comment:
I don't think this logic is quite right.  In the case that a user has a a default web app and a trial app, we should be using their setting.

I think I know why you're doing this and I think we can mitigate that by deleting the defaultWebAppId setting when we transfer the subscription IF it's the trial app.

Reply:
Ok, so to delete their setting we need to know their workspace. And the one place during transfer that we know their workspace is after we call getDeployFsPath within deploy. So somehow I need to know if they are transferring a trial app from within the deploy function.